utils/src/main/scala/za/co/absa/enceladus/utils/general/Section.scala

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.general
+
+import java.security.InvalidParameterException
+
+/**
+  * Represents a section of a string defined by its starting index and length of the section.
+  * It supports negative indexes for start, denoting a position counted from end
+  * _Comparison_
+  * The class implements the `Ordered` trait
+  * Sections with negative start are BIGGER then the ones with positive start
+  * The section with smaller ABSOLUTE value of start is considered smaller, if starts equal the shorter section is the
+  * smaller
+  * NB! Interesting consequence of the ordering is, that if the sections are not overlapping and removal applied on
+  *  string from greatest to smallest one by one, the result is the same as removing all sections "at once"
+  * @param start the start position of the section, if negative the position is counted from the end
+  * @param length length of the section, cannot be negative
+  */
+case class Section(start: Int, length: Int) extends Ordered[Section] {
+
+  if (length < 0) {
+    throw new InvalidParameterException(s"'length; cannot be negative, $length given")
+  }
+
+  if ((start >= 0) && (start.toLong + length.toLong > Int.MaxValue)) {
+    throw new IndexOutOfBoundsException(s"start and length combination are grater then ${Int.MaxValue}")
+  }
+
+  override def compare(that: Section): Int = {
+    if (start == that.start) {
+      length.compare(that.length) //shorter is smaller
+    } else if ((start < 0) && (that.start >= 0)) {
+      1 // negative start is bigger then the one of positive+zero
+    } else if ((start >= 0) && (that.start < 0)) {
+      -1
+    } else if (start.abs < that.start.abs) {
+      -1
+    } else {
+      1
+    }
+  }
+
+  /**
+    * Converts the Section to actual indexes representing the section on the given string if used in the substring function
+    * for example, e.g. first item of the tuple inclusive, second exclusive
```

Comment:
`for example` and `e.g.` mean the same thing, but I'm also confused by the example, shouldn't the example be more about where they can be used `e.g. as can be used in the substring function`, rather than explaining how `substring` works?